File: src/main/java/by/iba/repair_report/service/BranchService.java
package by.iba.repair_report.service;

import by.iba.repair_report.dto.request.BranchRequest;
import by.iba.repair_report.dto.response.BranchResponse;

import java.util.List;

public interface BranchService {
    BranchResponse createBranch(BranchRequest request);
    BranchResponse updateBranch(Long id, BranchRequest request);
    void deleteBranch(Long id);
    BranchResponse getBranchById(Long id);
    List<BranchResponse> getBranchesByEnterprise(Long enterpriseId);
    List<BranchResponse> getBranchesByAssociation(Long associationId);
    List<BranchResponse> getAllBranches();
    List<BranchResponse> searchBranches(Long enterpriseId, String name);
}

File: src/main/frontend/src/services/branchService.js
import api from './api';

export const branchService = {
    getAllBranches: async () => {
        const response = await api.get('/branches');
        return response.data;
    },

    getBranchById: async (id) => {
        const response = await api.get(`/branches/${id}`);
        return response.data;
    },

    createBranch: async (branchData) => {
        const response = await api.post('/branches', branchData);
        return response.data;
    },

    updateBranch: async (id, branchData) => {
        const response = await api.put(`/branches/${id}`, branchData);
        return response.data;
    },

    deleteBranch: async (id) => {
        await api.delete(`/branches/${id}`);
    },

    getBranchesByEnterprise: async (enterpriseId) => {
        const response = await api.get(`/branches/enterprise/${enterpriseId}`);
        return response.data;
    },

    getBranchesByAssociation: async (associationId) => {
        const response = await api.get(`/branches/association/${associationId}`);
        return response.data;
    },

    searchBranches: async (enterpriseId, name) => {
        const response = await api.get(`/branches/search?enterpriseId=${enterpriseId}&name=${name}`);
        return response.data;
    }
};